### PR TITLE
486 list doesnt get populated immediately when successfully submitting share permissions

### PR DESCRIPTION
**Pull request by arunapa**
https://user-images.githubusercontent.com/28983148/236089767-dd4d3aac-3217-499b-8ed2-0ebb863e36fd.mov

### CODE REVIEW

File: frontend/src/components/datasets/ShareDatasetModal.tsx [hunk 1]
```diff
@@ -46,9 +46,12 @@ export default function ShareDatasetModal(props: ShareDatasetModalProps) {
 	const [options, setOptions] = useState([]);
 	const users = useSelector((state: RootState) => state.group.users);
 
-	const setUserRole = (datasetId: string, username: string, role: string) =>
+	const setUserRole = async (datasetId: string, username: string, role: string) =>
```
Comment (longshuicy) [inline frontend/src/components/datasets/ShareDatasetModal.tsx:49]: oh why is the `async` needed

Comment (arunapa) [reply frontend/src/components/datasets/ShareDatasetModal.tsx:49]: To make sure that when the subsequent listRoles is called, the update is already reflected in the backend

Comment (longshuicy) [reply frontend/src/components/datasets/ShareDatasetModal.tsx:49]: ah sorry i didn't realize you are directly using dispatch :-) My bad it makes perfect sense now.

**Review by longshuicy — APPROVED** [review]
works well approved

**Review by max-zilla — APPROVED** [review]

**Pull request merged**